Term detail screen

fastlinemedia opened the pull request:
Resolves #172

fastlinemedia commented on src/system/ui/forms/index.js:
```diff
@@ -16,6 +16,7 @@ import {
 	SelectItem,
 	SuggestItem,
 	TaxonomyTermsItem,
+	ParentTermItems,
```
Is `ParentTermItems` item still being used? I'm not seeing a component for this and the term form is just using a standard select for the parent option.

shahidajmeri786 replied:
Yes it is used here:
![image](https://user-images.githubusercontent.com/50620598/72798650-8c390080-3c69-11ea-8810-cb0477d18a5c.png)

fastlinemedia commented on src/system/ui/forms/index.js:
```diff
@@ -96,6 +97,9 @@ Form.TextareaItem.displayName = 'Form.TextareaItem'
 Form.UrlItem.displayName = 'Form.UrlItem'
 
+Form.ParentTermItems = ParentTermItems
+Form.ParentTermItems.displayName = 'Form.ParentTermItems'
```
Same as above regarding `ParentTermItems`.

fastlinemedia commented on src/system/ui/forms/items/taxonomy-terms/index.js:
```diff
@@ -11,6 +11,7 @@ export const TaxonomyTermsItem = ( {
 	taxonomy,
 	value,
 	onChange,
+	isparentSelector,
```
Variable should be camelCase.

shahidajmeri786 replied:
Variable not needed anymore.

fastlinemedia commented on src/system/ui/forms/items/taxonomy-terms/index.js:
```diff
@@ -186,3 +187,63 @@ export const TaxonomyTermsItem = ( {
 }
+
+
+export const ParentTermItems = ( taxonomy,id ) => {
```
I'm seeing this now. Form items should be setup as components that can be used in the form config like so...

https://share.getcloudapp.com/qGuDBEKD

Let's break this out into its own file in `form/items` as well.

fastlinemedia commented on src/system/ui/pages/term/index.js:
```diff
@@ -1,40 +1,168 @@
+					title: {
+						label: __( 'Name' ),
+						labelPlacement: 'above',
+						value: title
```
Form config doesn't accept a value (you may have seen this in old code). To set the initial value, use the `defaults` object below.

fastlinemedia commented on src/system/ui/pages/term/index.js:
```diff
@@ -1,40 +1,168 @@
+					parent: {
+						label: __( 'Parent' ),
+						labelPlacement: 'above',
+						component: 'parent-terms',
```
Looking good! However, it doesn't look like that component was pushed as I'm getting an error when I try to build this.

fastlinemedia commented:
From closed PR...

**Parent Select**
Added the label placement be beside the select.
Added Only hierarchical terms should show the parent select.
Added the options show the hierarchy?

**Post Count**
Added the label placement be beside the count.

**Actions**
Added action buttons for:
View Archive
Edit in Admin
Delete

fastlinemedia commented:
Nice work @shahidajmeri786. Everything looks great. The only issue I'm seeing is a unique key error coming from the parent term items select...

https://share.getcloudapp.com/yAuLqn0y

Once that's fixed, we can merge this.

shahidajmeri786 commented:
I checked but I did not get any key error. 
![image](https://user-images.githubusercontent.com/50620598/73363363-9a70c780-42ce-11ea-86c4-f35d7e112842.png)

fastlinemedia commented:
> I checked but I did not get any key error.

I wonder if that error doesn't show if you don't have react dev tools installed? Anyhow, it comes from when you map an array to output elements (e.g. with select options). That is fixed by adding a key to each option like so...

https://share.getcloudapp.com/WnuEO5AJ

I pushed that fix and will get this merged. Thanks!
